Lots of objects in config results in Core 1 Panic'ed

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
<!--
Please provide details about your environment below this line. -->
Docker/hassio in Ubuntu X86

**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->
NodeMCU ESP32
 expressif ESP32-WROOM-32D

Tried with ESPhome 1.14.2, 1.14.3 and 1.15dev getting similar results with all of them
I also tried OTA and serial communication flashing, again, with similar results

The issue seems to be similar to #836 after the compile fix. I assumed that since 1.14.3 addressed that issue, it would also fix my problem, but it made no difference.


**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->
I'm fairly new to all this, not sure how to answer.
atm90e32 maybe?
I think you refer to the Circuit Setup 6 channel energy monitor board

**Description of problem:**
When uploading config YAML file with config for more than four power monitor boards the ESP32 enters a constant boot loop 

**Problem-relevant YAML-configuration entries:**
```yaml
PASTE YAML FILE HERE
substitutions:
# Change the disp_name to something you want  
  disp_name: south_panel
# Interval of how often the power is updated
  update_time: 6s
# 30A/1V SCT-013-030: 8650
# 50A/1V SCT-013-050: 15420
# 100A/50ma SCT-013-000: 32498
  current_cal: '36000'
#9VAC Transformer for versions > meter v1.2
  voltage_cal: '7305'
esphome:
  name: south_panel
  platform: ESP32
  board: nodemcu-32s

wifi:
  ssid: "WiFi"
  password: "password"

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "South Panel Fallback Hotspot"
    password: "password"

captive_portal:

# Enable logging
logger:

# Enable Home Assistant API
api:
  password: "password"

ota:
  password: "password"

web_server:
  port: 80

spi:
  clk_pin: 18
  miso_pin: 19
  mosi_pin: 23
sensor:
  - platform: wifi_signal
    name: ${disp_name} WiFi
    update_interval: 60s
#IC1 Main
  - platform: atm90e32
    cs_pin: 5
    phase_a:
      voltage:
        name: ${disp_name} Volts A Main
        id: ic1Volts
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT1 Amps
        id: ct1Amps
      power:
        name: ${disp_name} CT1 Watts 
        id: ct1Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT2 Amps 
        id: ct2Amps
      power:
        name: ${disp_name} CT2 Watts 
        id: ct2Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT3 Amps
        id: ct3Amps
      power:
        name: ${disp_name} CT3 Watts
        id: ct3Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq A Main
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC2 Main
  - platform: atm90e32
    cs_pin: 4
    phase_a:
      voltage:
        name: ${disp_name} Volts B Main
        id: ic2VoltsMain
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT4 Amps 
        id: ct4Amps
      power:
        name: ${disp_name} CT4 Watts 
        id: ct4Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT5 Amps 
        id: ct5Amps
      power:
        name: ${disp_name} CT5 Watts 
        id: ct5Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT6 Amps 
        id: ct6Amps
      power:
        name: ${disp_name} CT6 Watts 
        id: ct6Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq B
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC1 AddOn 1
  - platform: atm90e32
    cs_pin: 0
    phase_a:
      voltage:
        name: ${disp_name} Volts A AO1
        id: ic1VoltsAO1
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT7 Amps 
        id: ct7Amps
      power:
        name: ${disp_name} CT7 Watts
        id: ct7Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT8 Amps
        id: ct8Amps
      power:
        name: ${disp_name} CT8 Watts
        id: ct8Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT9 Amps
        id: ct9Amps
      power:
        name: ${disp_name} CT9 Watts
        id: ct9Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq A AO1
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC2 AddOn 1
  - platform: atm90e32
    cs_pin: 16
    phase_a:
      voltage:
        name: ${disp_name} Volts B AO1
        id: ic2VoltsAO1
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT10 Amps
        id: ct10Amps
      power:
        name: ${disp_name} CT10 Watts
        id: ct10Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT11 Amps
        id: ct11Amps
      power:
        name: ${disp_name} CT11 Watts
        id: ct11Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT12 Amps
        id: ct12Amps
      power:
        name: ${disp_name} CT12 Watts
        id: ct12Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq B AO1
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC1 AddOn 2
  - platform: atm90e32
    cs_pin: 2
    phase_a:
      voltage:
        name: ${disp_name} Volts A AO2
        id: ic1VoltsAO2
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT13 Amps 
        id: ct13Amps
      power:
        name: ${disp_name} CT13 Watts
        id: ct13Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT14 Amps
        id: ct14Amps
      power:
        name: ${disp_name} CT14 Watts
        id: ct14Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT15 Amps
        id: ct15Amps
      power:
        name: ${disp_name} CT15 Watts
        id: ct15Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq A AO2
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC2 AddOn 2
  - platform: atm90e32
    cs_pin: 17
    phase_a:
      voltage:
        name: ${disp_name} Volts B AO2
        id: ic2VoltsAO2
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT16 Amps
        id: ct16Amps
      power:
        name: ${disp_name} CT16 Watts
        id: ct16Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT17 Amps
        id: ct17Amps
      power:
        name: ${disp_name} CT17 Watts
        id: ct17Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT18 Amps
        id: ct18Amps
      power:
        name: ${disp_name} CT18 Watts
        id: ct18Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq B AO2
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC1 AddOn 3
  - platform: atm90e32
    cs_pin: 13
    phase_a:
      voltage:
        name: ${disp_name} Volts A AO3
        id: ic1VoltsAO3
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT19 Amps 
        id: ct19Amps
      power:
        name: ${disp_name} CT19 Watts
        id: ct19Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT20 Amps
        id: ct20Amps
      power:
        name: ${disp_name} CT20 Watts
        id: ct20Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT21 Amps
        id: ct21Amps
      power:
        name: ${disp_name} CT21 Watts
        id: ct21Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq A AO3
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC2 AddOn 3
  - platform: atm90e32
    cs_pin: 22
    phase_a:
      voltage:
        name: ${disp_name} Volts B AO3
        id: ic2VoltsAO3
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT22 Amps
        id: ct22Amps
      power:
        name: ${disp_name} CT22 Watts
        id: ct22Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT23 Amps
        id: ct23Amps
      power:
        name: ${disp_name} CT23 Watts
        id: ct23Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT24 Amps
        id: ct24Amps
      power:
        name: ${disp_name} CT24 Watts
        id: ct24Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq B AO3
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC1 AddOn 4
  - platform: atm90e32
    cs_pin: 14
    phase_a:
      voltage:
        name: ${disp_name} Volts A AO4
        id: ic1VoltsAO4
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT25 Amps 
        id: ct25Amps
      power:
        name: ${disp_name} CT25 Watts
        id: ct25Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT26 Amps
        id: ct26Amps
      power:
        name: ${disp_name} CT26 Watts
        id: ct26Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT27 Amps
        id: ct27Amps
      power:
        name: ${disp_name} CT27 Watts
        id: ct27Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq A AO4
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC2 AddOn 4
  - platform: atm90e32
    cs_pin: 25
    phase_a:
      voltage:
        name: ${disp_name} Volts B AO4
        id: ic2VoltsAO4
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT28 Amps
        id: ct28Amps
      power:
        name: ${disp_name} CT28 Watts
        id: ct28Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT29 Amps
        id: ct29Amps
      power:
        name: ${disp_name} CT29 Watts
        id: ct29Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT30 Amps
        id: ct30Amps
      power:
        name: ${disp_name} CT30 Watts
        id: ct30Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq B AO4
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC1 AddOn 5
  - platform: atm90e32
    cs_pin: 15
    phase_a:
      voltage:
        name: ${disp_name} Volts A AO5
        id: ic1VoltsAO5
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT31 Amps 
        id: ct31Amps
      power:
        name: ${disp_name} CT31 Watts
        id: ct31Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT32 Amps
        id: ct32Amps
      power:
        name: ${disp_name} CT32 Watts
        id: ct32Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT33 Amps
        id: ct33Amps
      power:
        name: ${disp_name} CT33 Watts
        id: ct33Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq A AO5
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#IC2 AddOn 5
  - platform: atm90e32
    cs_pin: 26
    phase_a:
      voltage:
        name: ${disp_name} Volts B AO5
        id: ic2VoltsAO5
        accuracy_decimals: 1
      current:
        name: ${disp_name} CT34 Amps
        id: ct34Amps
      power:
        name: ${disp_name} CT34 Watts
        id: ct34Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_b:
      current:
        name: ${disp_name} CT35 Amps
        id: ct35Amps
      power:
        name: ${disp_name} CT35 Watts
        id: ct35Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    phase_c:
      current:
        name: ${disp_name} CT36 Amps
        id: ct36Amps
      power:
        name: ${disp_name} CT36 Watts
        id: ct36Watts
      gain_voltage: ${voltage_cal}
      gain_ct: ${current_cal}
    frequency:
      name: ${disp_name} Freq B AO5
    line_frequency: 60Hz
    gain_pga: 1X
    update_interval: ${update_time}
#Total Amps Main
  - platform: template
    name: ${disp_name} Total Amps Main
    id: totalAmpsMain
    lambda: return id(ct1Amps).state + id(ct2Amps).state + id(ct3Amps).state + id(ct4Amps).state + id(ct5Amps).state + id(ct6Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}
#Total Amps AddOn
  - platform: template
    name: ${disp_name} Total Amps Add-on 1
    id: totalAmpsAddOn1
    lambda: return id(ct7Amps).state + id(ct8Amps).state + id(ct9Amps).state + id(ct10Amps).state + id(ct11Amps).state + id(ct12Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}      
#Total Amps AddOn 2
  - platform: template
    name: ${disp_name} Total Amps Add-on 2
    id: totalAmpsAddOn2
    lambda: return id(ct13Amps).state + id(ct14Amps).state + id(ct15Amps).state + id(ct16Amps).state + id(ct17Amps).state + id(ct18Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}      
#Total Amps AddOn 3
  - platform: template
    name: ${disp_name} Total Amps Add-on 3
    id: totalAmpsAddOn3
    lambda: return id(ct19Amps).state + id(ct20Amps).state + id(ct21Amps).state + id(ct22Amps).state + id(ct23Amps).state + id(ct24Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}
#Total Amps AddOn 4
  - platform: template
    name: ${disp_name} Total Amps Add-on 4
    id: totalAmpsAddOn4
    lambda: return id(ct25Amps).state + id(ct26Amps).state + id(ct27Amps).state + id(ct28Amps).state + id(ct29Amps).state + id(ct30Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}      
#Total Amps AddOn 5
  - platform: template
    name: ${disp_name} Total Amps Add-on 5
    id: totalAmpsAddOn5
    lambda: return id(ct31Amps).state + id(ct32Amps).state + id(ct33Amps).state + id(ct34Amps).state + id(ct35Amps).state + id(ct36Amps).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}      
#Total Amps
  - platform: template
    name: ${disp_name} Total Amps
    id: totalAmps
    lambda: return id(totalAmpsMain).state + id(totalAmpsAddOn1).state + id(totalAmpsAddOn2).state + id(totalAmpsAddOn3).state + id(totalAmpsAddOn4).state + id(totalAmpsAddOn5).state ;
    accuracy_decimals: 2
    unit_of_measurement: A
    icon: "mdi:flash"
    update_interval: ${update_time}
#Total Watts Main
  - platform: template
    name: ${disp_name} Total Watts Main
    id: totalWattsMain
    lambda: return id(ct1Watts).state + id(ct2Watts).state + id(ct3Watts).state + id(ct4Watts).state + id(ct5Watts).state + id(ct6Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}
#Total Watts AddOn1
  - platform: template
    name: ${disp_name} Total Watts Add-on1
    id: totalWattsAddOn1
    lambda: return id(ct7Watts).state + id(ct8Watts).state + id(ct9Watts).state + id(ct10Watts).state + id(ct11Watts).state + id(ct12Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}
#Total Watts AddOn2
  - platform: template
    name: ${disp_name} Total Watts Add-on2
    id: totalWattsAddOn2
    lambda: return id(ct13Watts).state + id(ct14Watts).state + id(ct15Watts).state + id(ct16Watts).state + id(ct17Watts).state + id(ct18Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}
#Total Watts AddOn3
  - platform: template
    name: ${disp_name} Total Watts Add-on3
    id: totalWattsAddOn3
    lambda: return id(ct19Watts).state + id(ct20Watts).state + id(ct21Watts).state + id(ct22Watts).state + id(ct23Watts).state + id(ct24Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}
#Total Watts AddOn4
  - platform: template
    name: ${disp_name} Total Watts Add-on4
    id: totalWattsAddOn4
    lambda: return id(ct25Watts).state + id(ct26Watts).state + id(ct27Watts).state + id(ct28Watts).state + id(ct29Watts).state + id(ct30Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}
#Total Watts AddOn5
  - platform: template
    name: ${disp_name} Total Watts Add-on5
    id: totalWattsAddOn5
    lambda: return id(ct31Watts).state + id(ct32Watts).state + id(ct33Watts).state + id(ct34Watts).state + id(ct35Watts).state + id(ct36Watts).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}
#Total Watts
  - platform: template
    name: ${disp_name} Total Watts
    id: totalWatts
    lambda: return id(totalWattsMain).state + id(totalWattsAddOn1).state + id(totalWattsAddOn2).state + id(totalWattsAddOn3).state + id(totalWattsAddOn4).state + id(totalWattsAddOn5).state ;
    accuracy_decimals: 1
    unit_of_measurement: W
    icon: "mdi:flash-circle"
    update_interval: ${update_time}
#kWh
  - platform: total_daily_energy
    name: ${disp_name} Total kWh
    power_id: totalWatts
    filters:
      - multiply: 0.001
    unit_of_measurement: kWh
switch:
  - platform: restart
    name: ${disp_name} Restart  
time:
  - platform: sntp
    id: sntp_time   


```

**Logs (if applicable):**
<!--
Please copy the debug log here. If possible, also connect to the ESP over USB and copy those logs into the backticks.
-->
```
Using 'COM3' as serial port.
Writing at 0x00094000... (100 %)Wrote 989648 bytes (548356 compressed) at 0x00010000 in 54.4 seconds (effective 145.6 kbit/s)...
Hash of data verified.

Leaving...
Hard Resetting...
Done! Flashing is complete!

Showing logs:
[23:14:54]ets Jun  8 2016 00:22:57
[23:14:54]
[23:14:54]rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:54]configsip: 0, SPIWP:0xee
[23:14:54]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:54]mode:DIO, clock div:2
[23:14:54]load:0x3fff0018,len:4
[23:14:54]load:0x3fff001c,len:952
[23:14:54]load:0x40078000,len:6084
[23:14:54]load:0x40080000,len:7944
[23:14:54]entry 0x40080310
[23:14:54]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:54]abort() was called at PC 0x4008caf8 on core 1
[23:14:54]
[23:14:54]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:54]
[23:14:54]Rebooting...
[23:14:54]ets Jun  8 2016 00:22:57
[23:14:54]
[23:14:54]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:54]configsip: 0, SPIWP:0xee
[23:14:54]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:54]mode:DIO, clock div:2
[23:14:54]load:0x3fff0018,len:4
[23:14:54]load:0x3fff001c,len:952
[23:14:54]load:0x40078000,len:6084
[23:14:54]load:0x40080000,len:7944
[23:14:54]entry 0x40080310
[23:14:55]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:55]abort() was called at PC 0x4008caf8 on core 1
[23:14:55]
[23:14:55]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:55]
[23:14:55]Rebooting...
[23:14:55]ets Jun  8 2016 00:22:57
[23:14:55]
[23:14:55]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:55]configsip: 0, SPIWP:0xee
[23:14:55]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:55]mode:DIO, clock div:2
[23:14:55]load:0x3fff0018,len:4
[23:14:55]load:0x3fff001c,len:952
[23:14:55]load:0x40078000,len:6084
[23:14:55]load:0x40080000,len:7944
[23:14:55]entry 0x40080310
[23:14:55]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:55]abort() was called at PC 0x4008caf8 on core 1
[23:14:55]
[23:14:55]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:55]
[23:14:55]Rebooting...
[23:14:55]ets Jun  8 2016 00:22:57
[23:14:55]
[23:14:55]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:55]configsip: 0, SPIWP:0xee
[23:14:55]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:55]mode:DIO, clock div:2
[23:14:55]load:0x3fff0018,len:4
[23:14:55]load:0x3fff001c,len:952
[23:14:55]load:0x40078000,len:6084
[23:14:55]load:0x40080000,len:7944
[23:14:55]entry 0x40080310
[23:14:55]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:55]abort() was called at PC 0x4008caf8 on core 1
[23:14:55]
[23:14:55]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:55]
[23:14:55]Rebooting...
[23:14:55]ets Jun  8 2016 00:22:57
[23:14:55]
[23:14:56]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:56]configsip: 0, SPIWP:0xee
[23:14:56]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:56]mode:DIO, clock div:2
[23:14:56]load:0x3fff0018,len:4
[23:14:56]load:0x3fff001c,len:952
[23:14:56]load:0x40078000,len:6084
[23:14:56]load:0x40080000,len:7944
[23:14:56]entry 0x40080310
[23:14:56]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:56]abort() was called at PC 0x4008caf8 on core 1
[23:14:56]
[23:14:56]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:56]
[23:14:56]Rebooting...
[23:14:56]ets Jun  8 2016 00:22:57
[23:14:56]
[23:14:56]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:56]configsip: 0, SPIWP:0xee
[23:14:56]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:56]mode:DIO, clock div:2
[23:14:56]load:0x3fff0018,len:4
[23:14:56]load:0x3fff001c,len:952
[23:14:56]load:0x40078000,len:6084
[23:14:56]load:0x40080000,len:7944
[23:14:56]entry 0x40080310
[23:14:56]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:56]abort() was called at PC 0x4008caf8 on core 1
[23:14:56]
[23:14:56]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:56]
[23:14:56]Rebooting...
[23:14:56]ets Jun  8 2016 00:22:57
[23:14:56]
[23:14:56]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:56]configsip: 0, SPIWP:0xee
[23:14:56]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:56]mode:DIO, clock div:2
[23:14:56]load:0x3fff0018,len:4
[23:14:56]load:0x3fff001c,len:952
[23:14:56]load:0x40078000,len:6084
[23:14:56]load:0x40080000,len:7944
[23:14:56]entry 0x40080310
[23:14:57]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:57]abort() was called at PC 0x4008caf8 on core 1
[23:14:57]
[23:14:57]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:57]
[23:14:57]Rebooting...
[23:14:57]ets Jun  8 2016 00:22:57
[23:14:57]
[23:14:57]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:57]configsip: 0, SPIWP:0xee
[23:14:57]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:57]mode:DIO, clock div:2
[23:14:57]load:0x3fff0018,len:4
[23:14:57]load:0x3fff001c,len:952
[23:14:57]load:0x40078000,len:6084
[23:14:57]load:0x40080000,len:7944
[23:14:57]entry 0x40080310
[23:14:57]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:57]abort() was called at PC 0x4008caf8 on core 1
[23:14:57]
[23:14:57]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:57]
[23:14:57]Rebooting...
[23:14:57]ets Jun  8 2016 00:22:57
[23:14:57]
[23:14:57]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:57]configsip: 0, SPIWP:0xee
[23:14:57]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:57]mode:DIO, clock div:2
[23:14:57]load:0x3fff0018,len:4
[23:14:57]load:0x3fff001c,len:952
[23:14:57]load:0x40078000,len:6084
[23:14:57]load:0x40080000,len:7944
[23:14:57]entry 0x40080310
[23:14:58]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:58]abort() was called at PC 0x4008caf8 on core 1
[23:14:58]
[23:14:58]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:58]
[23:14:58]Rebooting...
[23:14:58]ets Jun  8 2016 00:22:57
[23:14:58]
[23:14:58]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:58]configsip: 0, SPIWP:0xee
[23:14:58]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:58]mode:DIO, clock div:2
[23:14:58]load:0x3fff0018,len:4
[23:14:58]load:0x3fff001c,len:952
[23:14:58]load:0x40078000,len:6084
[23:14:58]load:0x40080000,len:7944
[23:14:58]entry 0x40080310
[23:14:58]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:58]abort() was called at PC 0x4008caf8 on core 1
[23:14:58]
[23:14:58]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:58]
[23:14:58]Rebooting...
[23:14:58]ets Jun  8 2016 00:22:57
[23:14:58]
[23:14:58]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:58]configsip: 0, SPIWP:0xee
[23:14:58]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:58]mode:DIO, clock div:2
[23:14:58]load:0x3fff0018,len:4
[23:14:58]load:0x3fff001c,len:952
[23:14:58]load:0x40078000,len:6084
[23:14:58]load:0x40080000,len:7944
[23:14:58]entry 0x40080310
[23:14:59]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:59]abort() was called at PC 0x4008caf8 on core 1
[23:14:59]
[23:14:59]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:59]
[23:14:59]Rebooting...
[23:14:59]ets Jun  8 2016 00:22:57
[23:14:59]
[23:14:59]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:59]configsip: 0, SPIWP:0xee
[23:14:59]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:59]mode:DIO, clock div:2
[23:14:59]load:0x3fff0018,len:4
[23:14:59]load:0x3fff001c,len:952
[23:14:59]load:0x40078000,len:6084
[23:14:59]load:0x40080000,len:7944
[23:14:59]entry 0x40080310
[23:14:59]***ERROR*** A stack overflow in task loopTask has been detected.
[23:14:59]abort() was called at PC 0x4008caf8 on core 1
[23:14:59]
[23:14:59]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:14:59]
[23:14:59]Rebooting...
[23:14:59]ets Jun  8 2016 00:22:57
[23:14:59]
[23:14:59]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:14:59]configsip: 0, SPIWP:0xee
[23:14:59]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:14:59]mode:DIO, clock div:2
[23:14:59]load:0x3fff0018,len:4
[23:14:59]load:0x3fff001c,len:952
[23:14:59]load:0x40078000,len:6084
[23:14:59]load:0x40080000,len:7944
[23:14:59]entry 0x40080310
[23:15:00]***ERROR*** A stack overflow in task loopTask has been detected.
[23:15:00]abort() was called at PC 0x4008caf8 on core 1
[23:15:00]
[23:15:00]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:15:00]
[23:15:00]Rebooting...
[23:15:00]ets Jun  8 2016 00:22:57
[23:15:00]
[23:15:00]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:15:00]configsip: 0, SPIWP:0xee
[23:15:00]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:15:00]mode:DIO, clock div:2
[23:15:00]load:0x3fff0018,len:4
[23:15:00]load:0x3fff001c,len:952
[23:15:00]load:0x40078000,len:6084
[23:15:00]load:0x40080000,len:7944
[23:15:00]entry 0x40080310
[23:15:00]***ERROR*** A stack overflow in task loopTask has been detected.
[23:15:00]abort() was called at PC 0x4008caf8 on core 1
[23:15:00]
[23:15:00]Backtrace: 0x4008c8b0:0x3ffaf780 0x4008cae1:0x3ffaf7a0 0x4008caf8:0x3ffaf7c0 0x40089bc4:0x3ffaf7e0 0x4008b828:0x3ffaf800 0x4008b7de:0xfefefefe
[23:15:00]
[23:15:00]Rebooting...
[23:15:00]ets Jun  8 2016 00:22:57
[23:15:00]
[23:15:00]rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
[23:15:00]configsip: 0, SPIWP:0xee
[23:15:00]clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
[23:15:00]mode:DIO, clock div:2
[23:15:00]load:0x3fff0018,len:4
[23:15:00]load:0x3fff001c,len:952
[23:15:00]load:0x40078000,len:6084
[23:15:00]load:0x40080000,len:7944
[23:15:00]entry 0x40080310
[23:15:00]
Serial port closed!


```

**Additional information and things you've tried:**
Since everything seemed to work fine with the basic connectivity config, I tried to upload the YAML file little by little. 
Everything works fine until I get over line 360 (sometimes 400, but most of the time line360).
If I try to upload anything over that, the ESP32 enters in a boot loop, with the console error detailed above and all connectivity is lost, until flashed again through serial.   

Please, do not hesitate to contact me if you have ANY question. 


<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

+1 for this issue, I am affected by this, in my case I needed to use both @jpeletier and @noderat for my project to work. Also, it is generating a lot of warnings of type:

```~/.platformio/packages/framework-arduinoespressif32@src-afacabbe58d012c05a6ee24950dc6ba1/tools/sdk/esp32/include/config/sdkconfig.h:68: warning: "CONFIG_ARDUINO_LOOP_STACK_SIZE" redefined
 #define CONFIG_ARDUINO_LOOP_STACK_SIZE 8192
```

--- a/esphome/codegen.py
+++ b/esphome/codegen.py
@@ -22,6 +22,7 @@ from esphome.cpp_generator import (  # noqa
     static_const_array,
     statement,
     variable,
+    with_local_variable,
     new_variable,
     Pvariable,
     new_Pvariable,

--- a/esphome/components/wifi/__init__.py
+++ b/esphome/components/wifi/__init__.py
@@ -332,8 +332,7 @@ def manual_ip(config):
     )
 
 
-def wifi_network(config, static_ip):
-    ap = cg.variable(config[CONF_ID], WiFiAP())
+def wifi_network(config, ap, static_ip):
     if CONF_SSID in config:
         cg.add(ap.set_ssid(config[CONF_SSID]))
     if CONF_PASSWORD in config:
@@ -360,14 +359,21 @@ async def to_code(config):
     var = cg.new_Pvariable(config[CONF_ID])
     cg.add(var.set_use_address(config[CONF_USE_ADDRESS]))
 
-    for network in config.get(CONF_NETWORKS, []):
+    def add_sta(ap, network):
         ip_config = network.get(CONF_MANUAL_IP, config.get(CONF_MANUAL_IP))
-        cg.add(var.add_sta(wifi_network(network, ip_config)))
+        cg.add(var.add_sta(wifi_network(network, ap, ip_config)))
+
+    for network in config.get(CONF_NETWORKS, []):
+        cg.with_local_variable(network[CONF_ID], WiFiAP(), add_sta, network)
 
     if CONF_AP in config:
         conf = config[CONF_AP]
         ip_config = conf.get(CONF_MANUAL_IP, config.get(CONF_MANUAL_IP))
-        cg.add(var.set_ap(wifi_network(conf, ip_config)))
+        cg.with_local_variable(
+            conf[CONF_ID],
+            WiFiAP(),
+            lambda ap: cg.add(var.set_ap(wifi_network(conf, ap, ip_config))),
+        )
         cg.add(var.set_ap_timeout(conf[CONF_AP_TIMEOUT]))
 
     cg.add(var.set_reboot_timeout(config[CONF_REBOOT_TIMEOUT]))

--- a/esphome/cpp_generator.py
+++ b/esphome/cpp_generator.py
@@ -5,7 +5,17 @@ import re
 from esphome.yaml_util import ESPHomeDataBase
 
 # pylint: disable=unused-import, wrong-import-order
-from typing import Any, Generator, List, Optional, Tuple, Type, Union, Sequence
+from typing import (
+    Any,
+    Callable,
+    Generator,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+    Sequence,
+)
 
 from esphome.core import (  # noqa
     CORE,
@@ -468,7 +478,9 @@ def statement(expression: Union[Expression, Statement]) -> Statement:
     return ExpressionStatement(expression)
 
 
-def variable(id_: ID, rhs: SafeExpType, type_: "MockObj" = None) -> "MockObj":
+def variable(
+    id_: ID, rhs: SafeExpType, type_: "MockObj" = None, register=True
+) -> "MockObj":
     """Declare a new variable, not pointer type, in the code generation.
 
     :param id_: The ID used to declare the variable.
@@ -485,8 +497,35 @@ def variable(id_: ID, rhs: SafeExpType, type_: "MockObj" = None) -> "MockObj":
         id_.type = type_
     assignment = AssignmentExpression(id_.type, "", id_, rhs)
     CORE.add(assignment)
-    CORE.register_variable(id_, obj)
+    if register:
+        CORE.register_variable(id_, obj)
     return obj
+
+
+def with_local_variable(
+    id_: ID, rhs: SafeExpType, callback: Callable[["MockObj"], None], *args
+) -> None:
+    """Declare a new variable, not pointer type, in the code generation, within a scoped block
+    The variable is only usable within the callback
+    The callback cannot be async.
+
+    :param id_: The ID used to declare the variable.
+    :param rhs: The expression to place on the right hand side of the assignment.
+    :param callback: The function to invoke that will receive the temporary variable
+    :param args: args to pass to the callback in addition to the temporary variable
+
+    """
+
+    # throw if the callback is async:
+    assert not inspect.iscoroutinefunction(
+        callback
+    ), "with_local_variable() callback cannot be async!"
+
+    CORE.add(RawStatement("{"))  # output opening curly brace
+    obj = variable(id_, rhs, None, True)
+    # invoke user-provided callback to generate code with this local variable
+    callback(obj, *args)
+    CORE.add(RawStatement("}"))  # output closing curly brace
 
 
 def new_variable(id_: ID, rhs: SafeExpType, type_: "MockObj" = None) -> "MockObj":

